Add options on interstitial 1

emmabeynon opened the pull request:
For [Trello](https://trello.com/c/KTbbRrYD/69-2-build-the-prototype-compare-different-ways-to-access-a-service-on-the-interstitial-page)

This is the first prototype for displaying sign-in options on the interstitial page for Check your state pension.

Outstanding actions
- [ ] Add missing copy (outlined in Trello card)

emmabeynon committed:
Add options-on-interstitial/check-state-pension-1 page

emmabeynon committed:
Add options-on-interstitial/check-state-pension-interstitial-1 page

We also add a route to handle the post request submitted by the form.

emmabeynon committed:
Add options-on-interstitial/create-new-account-1 page

emmabeynon committed:
Add options-on-interstitial/check-state-pension-1 to index page

nickcolley commented on app/routes.js:
```diff
@@ -7,5 +7,14 @@ router.get('/', function (req, res) {
 // add your routes here
+router.post('/check-state-pension-options-1', function (req, res) {
+  if (req.body.loginoption === "government-gateway") {
+    res.redirect('https://www.tax.service.gov.uk/gg/sign-in?continue=%2Fcheck-your-state-pension%2Faccount&origin=nisp-frontend&accountType=individual')
```
This isn't a pattern you see often, a form redirecting you to another domain, I wonder if there's any possibility for this to confuse users.

NickColley commented on app/views/options-on-interstitial/check-state-pension-1.html:
```diff
@@ -0,0 +1,118 @@
+{% extends "layout.html" %}
+
+{% block page_title %}
+  Start page example
```
Users of assistive tech use titles to know what page they have landed on, we probably need to think what this should be.

emmabeynon replied:
I've changed this to "Check your state pension".

NickColley commented on app/views/options-on-interstitial/check-state-pension-1.html:
```diff
@@ -0,0 +1,118 @@
+      <p>
+        Lorem ipsum dolor sit amet, te vix nisl splendide signiferumque, mel reque appetere cu. Pro eu populo conceptam. Id copiosae dissentias eam. Eum ea dico exerci dicunt, mea te habeo.
+      </p>
+      <a class="button" type="submit" href="./check-state-pension-interstitial-1">Sign in</a>
```
type="submit" is used on the `<input>` tag, here we could use `role="button"` instead to indicate that these are buttons.

emmabeynon replied:
I've changed this to `role="button"` 👍

NickColley commented on app/views/options-on-interstitial/check-state-pension-interstitial-1.html:
```diff
@@ -0,0 +1,41 @@
+    <div class="form-group">
+      <fieldset>
+        <div class="multiple-choice">
+          <input id="government-gateway" type="radio" name="loginoption" value="government-gateway"/>
```
At the moment when you interact with these radios, it's not clear what they're for.

To give context we need to move the heading so it's within the `fieldset` and within a `legend` this means when a user of assistive tech interacts with this form, the heading will be read out giving context to what they're doing.

emmabeynon replied:
I've had a go at addressing this - can you take a look to see if it's what you were expecting?

NickColley reviewed (commented):
.

emmabeynon committed:
Add error message to check-state-pension-interstitial-1

If a user does not select an option from the list of sign in options, we now
display an error requiring them to select an option.

NickColley committed:
Merge branch 'master' into add-options-on-interstitial-1
